### PR TITLE
Bug 1267134 - [Presentation WebAPI] Corresponding changes to align the new architecture of shell_remote.js for multi-screen.

### PR DESCRIPTION
**Pull request by kuoe0**
Hi @luke-chang, could you review this patch for me? I'll squash the commit after review passed. Thanks!

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5e0842570f68368ff35f4921892b1195d8cfbd46

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9908b53a8010fa9f89e4e25026d4f2753e8feca0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9908b53a8010fa9f89e4e25026d4f2753e8feca0

**Comment by KuoE0** [comment]
I left some comments on [Bug 1267134](https://bugzilla.mozilla.org/show_bug.cgi?id=1267134).

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9908b53a8010fa9f89e4e25026d4f2753e8feca0

File: apps/system/js/remote/remote_app_window_manager.js [hunk 1]
```diff
@@ -124,10 +123,10 @@
 
     _sendPresentationResult: function(detail) {
       this.debug(detail.type);
-      var evt = new CustomEvent('mozPresentationContentEvent', {
+      let evt = new CustomEvent('mozPresentationContentEvent', {
         bubbles: true,
         cancelable: false,
-        detail: detail
+        detail: detail,
```
Comment (luke-chang) [inline apps/system/js/remote/remote_app_window_manager.js]: redundant `,`?

File: apps/system/js/remote/remote_app_window_manager.js [hunk 2]
```diff
@@ -124,10 +123,10 @@
 
     _sendPresentationResult: function(detail) {
       this.debug(detail.type);
-      var evt = new CustomEvent('mozPresentationContentEvent', {
+      let evt = new CustomEvent('mozPresentationContentEvent', {
```
Comment (luke-chang) [inline apps/system/js/remote/remote_app_window_manager.js]: Although it's valid, I'm not sure why we need to use `let` here.

File: apps/system/js/remote/message_controller.js [hunk 3]
```diff
@@ -54,15 +54,16 @@
         ', ' + JSON.stringify(data.detail));
 
       switch(data.type) {
-        case 'launch-presentation-app':
-          this.debug('request to launchPresentationApp');
+        case 'app-config-ready':
           Service.request('launchPresentationApp', data.detail)
             .then(() => {
-              this.postMessage('launch-app-success', {
+              this.publish('launch-app-success', {
```
Comment (luke-chang) [inline apps/system/js/remote/message_controller.js]: I'm just curious who is the expecting receiver of the event `launch-app-success` and `launch-app-error`.

File: apps/system/js/remote/remote_app_window_manager.js (hunk 2 above)
Comment (KuoE0) [inline apps/system/js/remote/remote_app_window_manager.js]: I think we don't need to change this line. I'll remove this change!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4721b804fac2427a053c14c546c0b2b5854f5cf4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5874b38fe4c44706ff5e71fcd4a0a1e305843bb9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a2b43b8b5b857f29c36a85caa37821c83159e413

**Comment by KuoE0** [comment]
@luke-chang I updated the patch and test. Please review again, thank you!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e02c525eed6576d929dc0f43b5b7004d7cfbe546

**Pull request merged**